ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/PluginPhases.kt

```diff
@@ -0,0 +1,37 @@
+// ktlint-disable filename
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.application
+
+/**
+ * Descendents of [OnCallReceive] allow you to extend the process of sending a response to the client.
+ *
+ * Example:
+ *
+ * ```
+ * onCallRespond { call ->
+ *      println(call.request.uri)
+ * }
+ *
+ * onCallRespond.afterTransform { call, content ->
+ *      println("sending $content to a client")
+ * }
+ * ```
+ *
+ * This prints a URL once you execute call.respond() on your server
+ * and prints raw content to be sent to the client.
+ **/
+@PluginsDslMarker
+public interface OnCallRespond<PluginConfig : Any> {
```

Comment:
can be sealed interface?

Reply:
This can't be sealed - we inherit it our class

Reply:
but we inherit only in current package. The only thing, is that anonymous objects (as how it's used) can't inherit `sealed` interfaces/classes, so they should be converted to named objects/classes
BTW, missed `ApplicationPluginBuilder`, that also could be sealed. Same for it, it only implements by anonymous object

Reply:
I checked it, and it doesn't work in this case

Reply:
`ApplicationPluginBuilder` should be possible, and I will fix